mesonbuild/backend/backends.py

```diff
@@ -702,24 +690,24 @@ def create_test_serialisation(self, tests):
                 cmd = exe.get_command()
             else:
                 cmd = [os.path.join(self.environment.get_build_dir(), self.get_target_filename(t.get_exe()))]
-            is_cross = self.environment.is_cross_build() and \
-                self.environment.need_exe_wrapper()
-            if isinstance(exe, build.BuildTarget):
-                is_cross = is_cross and exe.is_cross
-            if isinstance(exe, dependencies.ExternalProgram):
+            if isinstance(exe, build.BuildTarget) or \
```

Comment:
It's possible enough classes have `for_machine` that these conditionals can go away.